enhancements/storage/csi-driver-global-configmap-secret-mounter.md

```diff
@@ -0,0 +1,201 @@
+---
+title: global-configmap-secret-mounter
+authors:
+  - "@janedoe"
+reviewers:
+  - TBD
+  - "@alicedoe"
+approvers:
+  - TBD
+  - "@oscardoe"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Global ConfigMap and Secret Mounter
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+ 1.  It is possible to build some sorts of per-namespace ACL boundaries if we really wanted to, but I think it makes more sense
+ to have the idea of global or namespaced and no in-between since nothing else is in between today.  They can write a controller
+ if they really want to create resources shared across some namespaces and not others.
+
+## Summary
+
+A CSI driver that can mount a configmap or secret from a specified namespace into any pod.
+This would allow a single copy of a configmap to exist in openshift-config-managed (say the trusted-ca) and mount that configmap into any pod.
+There is no ACL check on the pod attempting to mount it.  Any configmap or secret that can be mounted, can be mounted anywhere.
+The CSI driver must keep the content on disk up to date, similar to how the kubelet keeps configmap and secret data consistent today.
+This allows the creation of a webhook admission plugin to allow annotation based injection of well-known configmaps and secrets, like the trusted-ca today.
+
+To have reasonable security boundaries, the cluster-admin must explicitly enumerate all namespace/name tuples for the configmaps and secrets that he wants to expose. 
+This will allow the CSI driver to use individual list/watches to drive the watch behavior to have scoped RBAC for the CSI driver.
+
+## Motivation
+
+This CSI driver would generically solve the problem of "inject this custom CA bundle into every requesting pod".
+A clever admission webhook could even do this unconditionally across namespaces or even have a bash wrapping ability for system certs if so desired. 
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+
+### Non-Goals
+
+What is out of scope for this proposal? Listing non-goals helps to focus discussion
+and make progress.
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal actually is.
```

Comment:
Some examples would be great:
* Pod with a CSI volume that imports a secret from another namespace.
* Configuration of the CSI driver, especially the list of allowed config maps / secrets.
* (If part of this proposal) Pod before admission (incl. all annotations for the webhook).